using-timescaledb/compression.md

```diff
@@ -19,184 +19,237 @@ basics of setting this up for use in your environment.
+
+So, let’s consider a simplified example using a table that looks as follows:
+
+|time|device_id|cpu|disk_io|energy_consumption|
+|---|---|---|---|---|
+| 12:00:02 |1|88.2|20|0.8|
+| 12:00:01 |2|300.5|30| 0.9|
+| 12:00:01 |1|88.6|25|0.85|
+| 12:00:01 |2|299.1|40| 0.95|
+
+After converting this data to a single row, the data in “array” form:
+
+|time|device_id|cpu|disk_io|energy_consumption|
+|---|---|---|---|---|
+| [12:00:02, 12:00:02, 12:00:01, 12:00:1 ]| [1, 2, 1, 2]|[88.2, 300.5, 88.6, 299.1]|[20, 30, 25, 40] |[0.8, 0.9, 0.85, 0.95]| 
+
+#### Understanding the segmentby option [](understanding-segmentby)
+
+We can segment compressed rows by specific columns, so that each compressed
+row corresponds to data about a single item, e.g., a specific `device_id`.
+The `segmentby` option forces the system to break up the compressed array so
+that each compressed row has a single value for each segmentby column. For
+example if we set `device_id` as a `segmentby` column, then the compressed
+version of our running example would look like:
+
 
-So now your compressed data will look more like the following, which is divided
-into two segments (by `device_id`) and we can index `device_id` directly.
+|time|device_id|cpu|disk_io|energy_consumption|
+|---|---|---|---|---|
+| [12:00:02, 12:00:01]| 1 |[88.2, 88.6]|[20, 25] |[0.8, 0.85]| 
+| [12:00:02, 12:00:01]| 2 |[300.5, 299.1]|[30, 40] |[0.9, 0.95]| 
+ 
+The above example shows the the `device_id` column is no longer an array,
+instead it defines the single value associated with all of the compressed data in the row. 
 
-|device_id|time|value|
-|---|---|---|
-|1|8/22/2019 0:00|88.2|
-| |8/22/2019 1:00|88.6|
-| |8/22/2019 2:00|90.0|
-|2|8/22/2019 0:05|300.5|
-| |8/22/2019 1:05|302.0|
-| |8/22/2019 2:05|301.0|
+Because a single value is associated with a compressed row, no decompression
+is necessary to evaluate the value. Queries with WHERE clauses that filter by
+a `segmentby` column are much more efficient, as decompression can happen _after_
+filtering instead of before (thus avoiding the need to decompress
+filtered-out rows altogether). In fact, for even more efficient access,
+we build b-tree indexes over each `segmentby` column.
 
->:TIP: We do not recommend using `compress_segmentby` if you expect less than 100 values per
-segmenting column. If you don’t have enough rows per column value you won’t get
-ideal compression characteristics.
+`segmentby` columns are useful, but can be overused. If too many `segmentby` columns
+are specified, then the number of items in each compressed column becomes small
+and compression is not effective. Thus, we recommend that you make sure that
+each segment contains at least 100 rows per chunk. If this is not the case, 
+then you can move some segmentby columns into the orderby option (as described in the
+next section).
```

Comment:
Repetitive from right above?

Reply:
I think thats ok. we want people not to miss this (have already seen issues)